### PR TITLE
PYTHON-3299 Add Automatic Queryable Encryption Example to Docs

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/examples/encryption.rst [hunk 1]
```diff
@@ -336,6 +336,66 @@ data key and create a collection with the
...
+        ]
+      }
+  }
+
+  auto_encryption_opts = AutoEncryptionOpts(
+            kms_providers, key_vault_namespace, encrypted_fields_map=encrypted_fields_map)
+  client = MongoClient(auto_encryption_opts=auto_encryption_opts)
+  client.default.drop_collection('encryptedCollection')
+  coll = client.default.create_collection('encryptedCollection')
+  coll.insert_one({ "_id": 1, "firstName": "Snuffy"})
```
[the remark this answers is not in the document]
Comment (blink1073) [reply doc/examples/encryption.rst:394]: Done

File: doc/examples/encryption.rst [hunk 2]
```diff
@@ -336,6 +336,66 @@ data key and create a collection with the
...
+
+PyMongo 4.2 brings beta support for Queryable Encryption with MongoDB 6.0.
+
+Queryable Encryption is the second version of Client-Side Field Level Encryption.
+Data is encrypted client-side. Queryable Encryption supports indexed encrypted fields,
+which are further processed server-side.
+
+You must have MongoDB 6.0rc8+ Enterprise to preview the capability.
+
+Until PyMongo 4.2 release is finalized, it can be installed using::
```
[the remark this answers is not in the document]
Comment (blink1073) [reply doc/examples/encryption.rst:350]: Done

File: doc/examples/encryption.rst [hunk 3]
```diff
@@ -336,6 +336,66 @@ data key and create a collection with the
   if __name__ == "__main__":
       main()
 
+Automatic Queryable Encryption (Beta)
```
[the remark this answers is not in the document]
Comment (blink1073) [reply doc/examples/encryption.rst:339]: Done

**Review by ShaneHarvey — APPROVED** [review]
It’s a start!

**Review by juliusgeo — APPROVED** [review]
LGTM!

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

File: doc/examples/encryption.rst [hunk 4]
```diff
@@ -336,6 +336,79 @@ data key and create a collection with the
...
+            kms_providers, key_vault_namespace, encrypted_fields_map=encrypted_fields_map)
+  client = MongoClient(auto_encryption_opts=auto_encryption_opts)
+  client.default.drop_collection('encryptedCollection')
+  coll = client.default.create_collection('encryptedCollection')
+  coll.insert_one({ "_id": 1, "firstName": "Jane", "lastName": "Doe" })
+  docs = list(coll.find({"firstName": "Jane"}))
+  print(docs)
+
+In the above example, the ``firstName`` field is automatically encrypted and
+decrypted.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply doc/examples/encryption.rst]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**